File: src/main/java/com/jss/jssbackend/modules/miscellaneous/service/VatRateService.java
package com.jss.jssbackend.modules.miscellaneous.service;

import java.util.List;

import com.jss.jssbackend.modules.miscellaneous.model.VatRate;

public interface VatRateService {
    public List<VatRate> getVatRates();

    public VatRate getVatRate(Integer id);
}

File: client/src/app/modules/miscellaneous/services/vat.rate.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/appRest.service';
import { VatRate } from '../../miscellaneous/model/VatRate';

@Injectable({
  providedIn: 'root'
})
export class VatRateService extends AppRestService<VatRate>{

  constructor(http: HttpClient) {
    super(http, "miscellaneous");
  }

  getVatRates() {
    return this.getList(new HttpParams(), "vat-rates");
  }

}
